src/wolfsftp.c

```diff
@@ -2547,40 +2542,26 @@ static int wolfSSH_SFTPNAME_readdir(WOLFSSH* ssh, WDIR* dir, WS_SFTPNAME* out,
     out->fSz = sz;
 
     {
-        char* buf;
-        int   bufSz;
-        int   tmpSz;
+        char  r[WOLFSSH_MAX_FILENAME];
+        char  s[WOLFSSH_MAX_FILENAME];
 
-        bufSz = out->fSz + WSTRLEN(dirName) + sizeof(WS_DELIM);
-        buf = (char*)WMALLOC(bufSz + 1, out->heap, DYNTYPE_SFTP);
-        if (buf == NULL) {
-            return WS_MEMORY_E;
-        }
-        buf[0] = '\0';
+        r[0] = '\0';
         if (!special) { /* do not add dir name in special case */
-            WSTRNCAT(buf, dirName, bufSz + 1);
-            tmpSz = WSTRLEN(buf);
-
-            /* add delimiter between path and file/dir name */
-            if (tmpSz + 1 < bufSz) {
-                buf[tmpSz] = WS_DELIM;
-                buf[tmpSz+1] = '\0';
-            }
-
+            WSTRNCAT(r, dirName, sizeof(r));
```

Comment:
What if everything we're copying into r here doesn't fit? Won't r be garbled?

Reply:
Returning an error if everything won't fit into r, and freeing the out filename.